tests/unit/moduleapi/testrdb.tcl

```diff
@@ -128,5 +128,161 @@ tags "modules" {
+                    $master testrdb.set.before value1_master
+                    $replica testrdb.set.before value1_replica
+
+                    # Put different data sets on the master and replica
+                    # We need to put large keys on the master since the replica replies to info only once in 2mb
+                    $replica debug populate 2000 slave 10
+                    $master debug populate 1000 master 100000
+                    $master config set rdbcompression no
+
+                    # Force the replica to try another full sync (this time it will have matching master replid)
+                    $master multi
+                    $master client kill type replica
+                    $master set asdf asdf
+                    # the side effect of resizing the backlog is that it is flushed (16k is the min size)
+                    $master config set repl-backlog-size 16384
+                    $master exec
+
+                    # Set master with a slow rdb generation, so that we can easily intercept loading
+                    # 10ms per key, with 1000 keys is 5 seconds
+                    $master config set rdb-key-save-delay 5000
+                    
+                    # Wait for the replica to start reading the rdb
+                    wait_for_condition 100 100 {
+                        [s -1 async_loading] eq 1
+                    } else {
+                        fail "Replica didn't get into async_loading mode"
+                    }
+
+                    # Check that classic loading flag is NOT set to 1
+                    assert_equal [s -1 loading] 0
+                    
+                    # Check that after REDISMODULE_SUBEVENT_REPL_ASYNC_LOAD_STARTED, module global still has original replica value
+                    assert_equal value1_replica [$replica testrdb.get.before]
+
+                    # Make sure we're still async_loading to validate previous assertion
+                    assert_equal [s -1 async_loading] 1
+
+                    # Wait for loading to stop
+                    wait_for_condition 50 100 {
+                        [s -1 async_loading] eq 0
+                    } else {
+                        fail "Loading didn't stop"
+                    }
+
+                    # Check that after REDISMODULE_SUBEVENT_REPL_ASYNC_LOAD_COMPLETED, module key got value from master
+                    assert_equal value1_master [$replica testrdb.get.before]
+
+                    # Enable fast shutdown
+                    $master config set rdb-key-save-delay 0
+
+                    # Make sure amount of keys matches master
+                    assert_equal [$replica dbsize] 1001
+                }
+            }
+        } {} {external:skip}
+
+        test {Diskless load swapdb, module can use RedisModuleEvent_ReplAsyncLoad ABORTED event on aborted async_loading} {
```

Comment:
these two (or maybe soon to be 3) tests have a lot on common.
maybe we can run them in a `foreach` loop that executes the same code 3 times is a few `if`s to do the variations between them?
will also make it easier to realize what's different and make future adjustments.